lib/spack/spack/relocate.py

```diff
@@ -479,3 +479,111 @@ def substitute_rpath(orig_rpath, topdir, new_root_path):
         new_rpath = path.replace(topdir, new_root_path)
         new_rpaths.append(new_rpath)
     return new_rpaths
+
+
+def is_relocatable(spec):
+    """Returns True if an installed spec is relocatable.
+
+    Args:
+        spec (Spec): spec to be analyzed
+
+    Returns:
+        True if the binaries of an installed spec
+        are relocatable and False otherwise.
+
+    Raises:
+        ValueError: if the spec is not installed
+    """
+    if not spec.install_status():
+        raise ValueError('spec is not installed [{0}]'.format(str(spec)))
+
```

Comment:
This needs to check for virtual or external before attempting to check the files. I added 
    if spec.external or spec.virtual:
        return False